#!/bin/bash
#PBS -S /bin/bash

## I want 4 processors 
#PBS -l procs=4

## How much RAM does each processor need?
#PBS -l pmem=2000mb

## The maximum walltime that will be used for my job 
#PBS -l walltime=04:15:00

## I want email sent when the job begins, ends and aborts (bea)
#PBS -m bea

## Where I want the email to be sent
#PBS -M YourEmailAddress@gmail.com 

## Let's run this job from the place that we submit the script from 
cd $PBS_O_WORKDIR

## Define variables to be used in the commands below
WORKING_DIR='/home/richmonp/Cheung2017/PROCESS/'
GENOME='/home/richmonp/Cheung2017/GENOME/S288c_BWA_Index'
FASTQR1='/home/richmonp/Cheung2017/RAW_DATA/YM2449-3*fastq.gz'
SAM='YM2449-3_BWA.sam'
BAM='YM2449-3_BWA.bam'
SORTEDBAM='YM2449-3_BWA.sorted.bam'
SORTEDBAMPREFIX='YM2449-3_BWA.sorted'
FLAGSTAT='YM2449-3_BWA.stats'

## Load the module containing bwa and samtools
module load bio-tools
module load bio-tools/WG_Genomic_Analysis_Workshop

## Let's go to directory we want to work in
cd $WORKING_DIR

## Map with BWA
bwa mem -t 4  $GENOME \
$FASTQR1 $FASTQR2 \
> $WORKING_DIR$SAM

## Convert sam to bam using samtools view
samtools view  -b $WORKING_DIR$SAM -o $WORKING_DIR$BAM 

## Sort the bam file using samtools sort
samtools sort $WORKING_DIR$BAM $WORKING_DIR$SORTEDBAMPREFIX 

## Index the sorted bam 
samtools index $WORKING_DIR$SORTEDBAM  

# Get the stats
samtools flagstat $WORKING_DIR$SORTEDBAM > $WORKING_DIR$FLAGSTAT

